### PR TITLE
4642 fix pipenv lock --pre doesn't include prerelease dependencies

### PR DESCRIPTION
**Pull request by nid5**
### The issue

#4642 
pipenv lock --pre doesn't actually include prerelease dependencies

### The fix

Used the already existing self.pre variable to append --pre to a built command that ultimately results in prerelease dependencies being included in the locking process. This seems to be consistent with conventions at least within that file, and doesn't require additional fields to be defined.

### The checklist

* [x] Associated issue
* [x] A news fragment in the `news/` directory to describe this fix with the extension `.bugfix`, `.feature`, `.behavior`, `.doc`. `.vendor`. or `.trivial` (this will appear in the release changelog). Use semantic line breaks and name the file after the issue number or the PR #.

### CODE REVIEW

File: pipenv/utils.py [hunk 1]
```diff
@@ -697,24 +697,27 @@ def pip_command(self):
...
-    def prepare_pip_args(self, use_pep517=False, build_isolation=True):
+    def prepare_pip_args(self, use_pep517=False, build_isolation=True, pre=False):
         pip_args = []
         if self.sources:
             pip_args = prepare_pip_source_args(self.sources, pip_args)
         if use_pep517 is False:
             pip_args.append("--no-use-pep517")
         if build_isolation is False:
             pip_args.append("--no-build-isolation")
+        if pre is True:
```
Comment (uranusjr) [inline pipenv/utils.py:708]:
```suggestion
        if self.pre:
```

So you don’t need to pass the value around.

**Review by uranusjr — CHANGES_REQUESTED** [review]

File: pipenv/utils.py [hunk 2]
```diff
@@ -697,24 +697,27 @@ def pip_command(self):
...
 
     @property
     def pip_args(self):
+        pre = self.pre
         use_pep517 = environments.get_from_env("USE_PEP517", prefix="PIP")
         build_isolation = environments.get_from_env("BUILD_ISOLATION", prefix="PIP")
         if self._pip_args is None:
             self._pip_args = self.prepare_pip_args(
-                use_pep517=use_pep517, build_isolation=build_isolation
+                use_pep517=use_pep517, build_isolation=build_isolation, pre=pre
```
Comment (frostming) [inline pipenv/utils.py:720]: So we can drop the `pre` argument now since it isn't used any more?

Comment (nid5) [reply pipenv/utils.py]: New commit removes unnecessary argument.

**Review by frostming — APPROVED** [review]

**Review by uranusjr — APPROVED** [review]

**Comment by frostming** [comment]
Sorry to miss the updates here, will merge after the CI finishes

**Pull request merged**